src/builder.py

```diff
@@ -121,44 +126,25 @@ def install_clouds_yaml(cloud_config: dict) -> None:
         OPENSTACK_CLOUDS_YAML_PATH.write_text(yaml.safe_dump(cloud_config), encoding="utf-8")
 
 
-def configure_cron(run_config: state.BuilderRunConfig, interval: int) -> bool:
+def configure_cron(unit_name: str, interval: int) -> bool:
     """Configure cron to run builder.
 
     Args:
-        run_config: The configuration required to run builder.
+        unit_name: The charm unit name to run cronjob dispatch hook.
         interval: Number of hours in between image build runs.
 
     Returns:
         True if cron is reconfigured. False otherwise.
     """
     commands = [
-        # HOME path is required for GO modules.
-        f"HOME={UBUNTU_HOME}",
         "/usr/bin/run-one",
-        "/usr/bin/sudo",
-        "--preserve-env",
-        str(GITHUB_RUNNER_IMAGE_BUILDER_PATH),
-        "run",
-        run_config.cloud_name,
-        IMAGE_NAME_TMPL.format(
-            IMAGE_BASE=run_config.base.value,
-            ARCH=run_config.arch.value,
-        ),
-        "--base-image",
-        run_config.base.value,
-        "--keep-revisions",
-        str(run_config.num_revisions),
-        "--callback-script",
-        str(run_config.callback_script.absolute()),
+        "/usr/bin/bash",
+        "-c",
+        f'/usr/bin/juju-exec "{unit_name}" "JUJU_DISPATCH_PATH=run HOME={UBUNTU_HOME} ./dispatch"',
```

Comment:
One thing to keep in mind is that we have observed a strange problem where juju-exec can hang on upgrade: https://bugs.launchpad.net/juju/+bug/2055184

We are not using a systemd timer here, but cron instead, so this should not have the same effect as described in the bug. Also, the command won't run as often as in the github-runner charm (reconciliation is usually a few minutes, not hours), so it's pretty unlikely we'll run into this problem. But I wanted to mention it anyway (that we might have a case where a process is not cleaned up and hangs).

Reply:
Thank you! I wasn't aware of it!